### ISSUE TITLE
Improve SurveyPage accessibility

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#19219]
Survey page: https://sourcegraph.com/survey

This page has poor accessibility for screen reader and keyboard users.

## Known issues:
1. The score field should be rendered as a list of radio buttons, rather than a list of links.

Score field:
![image](https://user-images.githubusercontent.com/9516420/111486433-64d7a980-872f-11eb-8ab4-f6d316cb766e.png)
